manifests/init.pp

```diff
@@ -18,7 +18,19 @@
 # $plugin_version::             foreman plugins version, it's passed to ensure parameter of plugins package resource
 #                               can be set to 'latest', 'present',  'installed', 'absent'.
 #
-# $port::                       Port on which will foreman proxy listen
+# $port::                       Port to listen on (deprecated in favor of $https_port and $http_port)
+#                               type:integer
+#
+# $http::                       Enable HTTP
+#                               type:boolean
+#
+# $http_port::                  HTTP port to listen on (if http is enabled)
+#                               type:integer
+#
+# $ssl::                        Enable SSL, ensure proxy is added with "https://" protocol if true
+#                               type:boolean
+#
+# $https_port:                  HTTPS port to listen on (if ssl is enabled)
```

Comment:
I don't have a strong view about "https" versus "ssl" in this context, but I do feel that the parameter names should be similar to each other.  If you're looking in --help and see one, you should be able to find the other near it.